src/core/catalog.js

```diff
@@ -638,10 +639,39 @@ class Catalog {
       return { name: stringToPDFString(nestedName), order: nestedOrder };
     }
 
+    function parseRBGroups(rbGroups) {
+      if (!Array.isArray(rbGroups)) {
+        return;
+      }
+
+      for (const value of rbGroups) {
+        const rbGroup = xref.fetchIfRef(value);
+        if (!Array.isArray(rbGroup) || !rbGroup.length) {
+          continue;
+        }
+
+        const parsedRbGroup = new Set();
+
+        for (const ref of rbGroup) {
+          if (
+            ref instanceof Ref &&
+            groupRefCache.has(ref) &&
+            !parsedRbGroup.has(ref)
```

Comment:
Unfortunately I don't believe that this last check works, since `parsedRbGroup` contains strings while `ref instanceof Ref` here.

Reply:
I cannot imagine that the rather random pixel discrepancies of the failed testcases have anything to do with the added OC-related code of the patch. Only the PDF of issue2442 contains an `OCProperties` dictionary. While it has an `RBGroups` sub-key, the value thereof is an empty array. So, even here, the code listed above is never reached and thus cannot have an impact on the rendering. Given that the discrepancies of colour components are not larger than `1`, they might be caused by round-off error. Are, perhaps, the reference images outdated?

Reply:
There's unfortunately a number of intermittent issues in the reference tests, so that's usually not anything to worry about and thus *not* what I commented on above!

---

The point here is that there's a logical "error" in the patch, since `!parsedRbGroup.has(ref) === true` will always hold and that check is thus pointless as-is. Did you actually mean the following?
```suggestion
            !parsedRbGroup.has(ref.toString())
```

Reply:
Oh, apologies, I completely misread your former comment! You are completely right, the test as of now does not make sense at all. Of course, `ref.toString()` is needed here! Maybe, it would have been safer to use `RefSet` instead of `Set` for `parsedRbGroup`, where conversion to string is internally handled. But I don't now how to iterate over `RefSet` values in a `for...of` loop. This is needed in `src/display/optional_content_config.js`, line 245. 

How can I make use of the `Symbol.iterator` defined in class `RefSet`?